src/test/java/uk/gov/mca/beacons/service/registrations/RegistrationsControllerIntegrationTest.java

```diff
@@ -15,29 +20,70 @@
 class RegistrationsControllerIntegrationTest {
 
   private static final String REGISTRATION_ENDPOINT = "/registrations/register";
+  private static final String REGISTRATION_JSON_RESOURCE =
+    "src/test/resources/fixtures/registrations.json";
+  private static final ObjectMapper OBJECT_MAPPER = new ObjectMapper();
 
   @Autowired
   private WebTestClient webTestClient;
 
-  @Test
-  void givenNewValidRegistration_whenPosted_thenStatus201() throws IOException {
-    final String validRegistrationRequestBody = importFromFile(
-      "src/test/resources/validRegistration.json"
-    );
+  @ParameterizedTest
```

Comment:
We are using [JUnit 5](https://junit.org/junit5/docs/current/user-guide/) as our testing library for Spring Boot - this is the standard library for Java.

The docs above are worth a good look through.  The [Paramaterized Test](https://junit.org/junit5/docs/current/user-guide/#writing-tests-parameterized-tests) annotation allows us to run the same test multiple times with different inputs.  

Here we are just testing the valid/invalid JSON payloads to the REST controller.  This is calling through to the services and persisting the beacons within the database.

I haven't currently that the values within the JSON are what has been stored within the database as I'm not entirely convinced we should be returning the full registration object back.  My gut feel is to return the reference back to the consumer, which we could then look to use within this test to assert objects with certain fields have been persisted.

Reply:
I think the "RESTful" thing to do is return the URI of the POSTed resource.  Check out JSONAPI's [standard for POSTing](https://jsonapi.org/format/#crud-creating) (I really think we should follow this, as @NyWeb mentioned):

```
HTTP/1.1 201 Created
Location: http://example.com/photos/550e8400-e29b-41d4-a716-446655440000
Content-Type: application/vnd.api+json

{
  "data": {
    "type": "photos",
    "id": "550e8400-e29b-41d4-a716-446655440000",
    "attributes": {
      "title": "Ember Hamster",
      "src": "http://example.com/images/productivity.png"
    },
    "links": {
      "self": "http://example.com/photos/550e8400-e29b-41d4-a716-446655440000"
    }
  }
}
```

This suggests we should persist the `Registration` object, which is different to our current thinking (see previous comment mentioning not persisting a `Registration`, only its constituent parts)`. (I'd agree that we should persist `Registration`: if the semantic meaning of a POST request is 'create resource', we should persist and allow GETting of that resource.)